ownCloud File Provider/FileProviderExtension.m

```diff
@@ -50,6 +51,30 @@ - (instancetype)init
 	OCAppIdentity.sharedAppIdentity.keychainAccessGroupIdentifier = bundleInfoDict[@"OCKeychainAccessGroupIdentifier"];
 	OCAppIdentity.sharedAppIdentity.appGroupIdentifier = bundleInfoDict[@"OCAppGroupIdentifier"];
 
+	// Initialize crash reporter as soon as FP extension is loaded into memory
+	PLCrashReporterConfig *configuration = [PLCrashReporterConfig defaultConfiguration];
+	PLCrashReporter *reporter = [[PLCrashReporter alloc] initWithConfiguration:configuration];
+
+	// Do we have a pending crash report from previous session?
+	if ([reporter hasPendingCrashReport]) {
+
+		// Generate a report and add it to the log file
+		NSData *crashData = [reporter loadPendingCrashReportData];
+		if (crashData != nil) {
+			PLCrashReport *report = [[PLCrashReport alloc] initWithData:crashData error:nil];
+			if (report != nil) {
+				NSString *crashString = [PLCrashReportTextFormatter stringValueForCrashReport:report withTextFormat:PLCrashReportTextFormatiOS];
+				OCTLogError(@[@"CRASH_REPORTER"], @"%@", crashString);
+			}
+		}
+
+		// Purge the report which we just added to the log
+		[reporter purgePendingCrashReport];
+	}
+
+	// Start intercepting OS signals to catch crashes
+	[reporter enableCrashReporter];
```

Comment:
I haven't been able to provoke the generation of a crash log, but maybe that's due my testing setup and/or Xcode involvement.

I think the new location in `-[FileProviderExtension init]` could run into a problem, however, since `enableCrashReporter` may only be called once, or the crash reporter will crash by raising an exception in `PLCrashReporter.m:577`.

Since there can be multiple instances of `FileProvider` (one per account AFAIK) in a single process, this would become a problem.

Placing it in `+load` or making sure, via `dispatch_once` (using a static variable as backing) or similar, that this code is only run once, would therefore be important.

Since you know how to provoke a loggable crash already, and can try, I didn't make that change. Before noticing the `enableCrashReporter` issue, I did resolve merge conflicts and added a `CRASH_REPORTER` tag to the line logging the crash report, so that it's logged just like from the app.